pkg/bucketeer/sdk.go

```diff
@@ -221,6 +253,49 @@ func (s *sdk) BoolVariation(ctx context.Context, user *user.User, featureID stri
 	return v
 }
 
+func (s *sdk) BoolVariationDetail(
```

Comment:
To avoid duplicating code and possible inconsistencies, we should call the `BoolVariationDetails` in the `BoolVariation`.
The same for all the interfaces.

E.g.

```go
func (s *sdk) BoolVariation(ctx context.Context, user *user.User, featureID string, defaultValue bool) bool {
	return s.BoolVariationDetail(ctx, user, featureID, defaultValue).Value
}
```

https://github.com/bucketeer-io/go-server-sdk/pull/136/files#diff-9658208f158246a9cfbc2c798e77e8224eaed053c42d140eb1e7237d8ed4e91cR239-R252